packages/app-utils/src/components/AppBootstrap.test.tsx

```diff
@@ -0,0 +1,151 @@
+import React from 'react';
+import { AUTH_HANDLER_TYPE_ANONYMOUS } from '@deephaven/auth-plugins';
+import { ApiContext } from '@deephaven/jsapi-bootstrap';
+import {
+  CoreClient,
+  IdeConnection,
+  dh as DhType,
+} from '@deephaven/jsapi-types';
+import { TestUtils } from '@deephaven/utils';
+import { act, render, screen } from '@testing-library/react';
+import AppBootstrap from './AppBootstrap';
+
+const API_URL = 'http://mockserver.net:8111';
+const PLUGINS_URL = 'http://mockserver.net:8111/plugins';
+
+const mockPluginsPromise = Promise.resolve([]);
+jest.mock('../plugins', () => ({
+  ...jest.requireActual('../plugins'),
+  loadModulePlugins: jest.fn(() => mockPluginsPromise),
+}));
+
+const mockChildText = 'Mock Child';
+const mockChild = <div>{mockChildText}</div>;
+
+function expectMockChild() {
+  return expect(screen.queryByText(mockChildText));
+}
+
+it('should throw if api has not been bootstrapped', () => {
+  expect(() =>
+    render(
+      <AppBootstrap apiUrl={API_URL} pluginsUrl={PLUGINS_URL}>
+        {mockChild}
+      </AppBootstrap>
+    )
+  ).toThrow();
+  expectMockChild().toBeNull();
+});
+
+it('should display an error if no login plugin matches the provided auth handlers', async () => {
+  const authConfigValues: [string, string][] = [
+    ['AuthHandlers', `MockAuthHandler`],
+  ];
+  const mockGetAuthConfigValues = jest.fn(() =>
+    Promise.resolve(authConfigValues)
+  );
+  const mockLogin = jest.fn(() => Promise.resolve());
+  const client = TestUtils.createMockProxy<CoreClient>({
+    getAuthConfigValues: mockGetAuthConfigValues,
+    login: mockLogin,
+  });
+  const api = TestUtils.createMockProxy<DhType>({
+    CoreClient: (jest
+      .fn()
+      .mockImplementation(() => client) as unknown) as CoreClient,
+  });
+
+  render(
+    <ApiContext.Provider value={api}>
+      <AppBootstrap apiUrl={API_URL} pluginsUrl={PLUGINS_URL}>
+        {mockChild}
+      </AppBootstrap>
+    </ApiContext.Provider>
+  );
+  expectMockChild().toBeNull();
+  expect(mockGetAuthConfigValues).toHaveBeenCalled();
+
+  await act(async () => {
+    await mockPluginsPromise;
+  });
+
+  expectMockChild().toBeNull();
+  expect(mockLogin).not.toHaveBeenCalled();
+  expect(
+    screen.queryByText(
+      'Error: No login plugins found, please register a login plugin for auth handlers: MockAuthHandler'
+    )
+  ).not.toBeNull();
```

Comment:
You can use `toBeInTheDocument()` to test these instead if you want, but this is also fine

Reply:
I always forget about that one for some reason

Reply:
Also weird - my TS doesn't seem to be happy with it?? Even in our existing usages:
![image](https://user-images.githubusercontent.com/4505624/233484755-9052f62b-1393-4a58-a65d-210beae44060.png)

I'll just leave this for this change, I'll need to figure out why my computer is confused.

Reply:
Might be a tsconfig/tsserver issue or something. It's an ambient type that is from `@testing-library/jest-dom`

Reply:
Hmm yea I restarted the TS server a couple times and it was still giving me the error, looks like I just needed to wait longer though... was looking at something else for a few minutes, then when I came back it was all good with it.